tla/tlc.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+#
+# Python TLC wrapper script for the CCF project
+# Goals:
+# - No dependencies, no venv, no pip install
+# - Set sensible defaults with an eye on performance
+# - Capture useful switches for CI, debugging
+# - Expose specification configuration through CLI
+# - Provide a useful --help, and basic sanity checks
+
+import os
+import argparse
+import pprint
+import pathlib
+
+DEFAULT_JVM_ARGS = [
+    "-XX:+UseParallelGC",  # Use parallel garbage collection, for performance
+]
+
+
+DEFAULT_CLASSPATH_ARGS = ["-cp", "tla2tools.jar:CommunityModules-deps.jar"]
+
+
+def cli():
+    parser = argparse.ArgumentParser(
+        description="TLC model checker wrapper for the CCF project"
+    )
+
+    # Common options for all commands
+    parser.add_argument(
+        "-v",
+        action="store_true",
+        help="Print out command and environment before running",
+    )
+    # Changes to TLC defaults
+    parser.add_argument(
+        "--disable-cdot", action="store_true", help="Disable \\cdot support"
+    )
+    parser.add_argument(
+        "--workers",
+        type=str,
+        default="auto",
+        help="Number of workers to use, default is 'auto'",
+    )
+    parser.add_argument(
+        "--checkpoint", type=int, default=0, help="Checkpoint interval, default is 0"
+    )
+    parser.add_argument(
+        "--lncheck",
+        type=str,
+        choices=["final", "default"],
+        default="final",
+        help="Liveness check, set to 'default' to run periodically or 'final' to run once at the end, default is final",
+    )
+    # Convenient shortcuts applicable to all commands
+    parser.add_argument(
+        "--jmx",
+        action="store_true",
+        help="Enable JMX to allow monitoring, use echo 'get -b tlc2.tool:type=ModelChecker CurrentState' | jmxterm -l localhost:55449 -i to query",
+    )
+    parser.add_argument(
+        "--dot", action="store_true", help="Generate a dot file for the state graph"
+    )
+    parser.add_argument(
+        "--trace-name",
+        type=str,
+        default=None,
+        help="Name to give to the trace files, defaults to base name of the spec, or config if provided",
+    )
+    parser.add_argument(
+        "--config",
+        type=pathlib.Path,
+        default=None,
+        help="Path to the TLA+ configuration",
+    )
+
+    subparsers = parser.add_subparsers(dest="cmd")
+
+    # Model checking
+    mc = subparsers.add_parser("mc", help="Model checking")
```

Comment:
is there someway to express that this mode is supported by both the consistency and consensus specs but the arguments are consensus only?

Reply:
Nothing obvious comes to mind, that's the problem with env vars, you can always set them, but if the process/spec doesn't do anything with them you can't really tell.